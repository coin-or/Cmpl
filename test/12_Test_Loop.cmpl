
const:
	echo "Einfache for-Schleife";
	{ i := 1..3: echo i; }

	echo "while-Schleife";
	a := 1;
	{ a < 5: echo a; a := a + 1; repeat; }

	echo "for-Schleife, die mehrfach gestartet wird";
	a := 1;
	{ i := 1..3: echo a + ": " + i; a := a + 1; { a <= 3: repeat i; } }

	echo "Aussere while-Schleife mit innerer for-Schleife";
	a := 1;
	xx {:
		{ i := 1..3: echo a + ": " + i; }
		a := a + 1;
		{ a <= 3: repeat xx; }

	}

	echo "Mehrfach gestartete innere for-Schleife, mit Alternative";
	a := 3;
	{ i := a(-1)1:
		echo a + ": " + i;
		{ i = 1: a := a - 1; repeat i; }
	  | default: echo "Ende";
	}

	echo "Definition einer lokalen Konstante";
	{ k := 1.6: echo k; }

	echo "Mehrfach gestartete Definition einer lokalen Konstante";
	a := 3;
	{ k := a, k >= 1:
		echo k;
		a := a - 1;
		repeat k;
	}

	echo "while-Schleife ueber lokal definierte Konstante";
	{ k := 3, k >= 1:
		echo k;
		k := k - 1;
		repeat;
	}

	echo "for-Schleife mit Zuweisung an die Laufkonstante";
	s := 10(-2)1;
	{ k := 1, i in s:
		echo i;
		{ k && i = 2: k := 0; i := 6; }
		echo i;
	}

