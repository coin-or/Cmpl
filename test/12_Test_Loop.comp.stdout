Einfache for-Schleife
1
2
3
while-Schleife
1
2
3
4
for-Schleife, die mehrfach gestartet wird
1: 1
2: 1
3: 1
4: 2
5: 3
Aussere while-Schleife mit innerer for-Schleife
1: 1
1: 2
1: 3
2: 1
2: 2
2: 3
3: 1
3: 2
3: 3
Mehrfach gestartete innere for-Schleife, mit Alternative
3: 3
3: 2
3: 1
2: 2
2: 1
1: 1
Ende
Definition einer lokalen Konstante
1.600000
Mehrfach gestartete Definition einer lokalen Konstante
3
2
1
while-Schleife ueber lokal definierte Konstante
3
2
1
for-Schleife mit Zuweisung an die Laufkonstante
10
10
8
8
6
6
4
4
2
6
4
4
2
2
